### PR TITLE
Fixed appengine starting on the same port.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

File: AppController/djinn.rb [hunk 1]
```diff
@@ -5015,12 +5040,17 @@ def get_scaling_info_for_app(app_name, update_dashboard=true)
   #     occurs when we start the app or add/remove AppServers).
   #   time_requests_were_seen: An Integer that represents the epoch time when we
   #     got request info from haproxy.
+  #   total_req_in_queue: An Integer that represents the current number of
+  #     requests waiting to be served.
+  #   update_dashboard: a boolean to indicate if we send the information
```
Comment (menivaitsi) [inline AppController/djinn.rb]: a boolean -> A boolean

**Comment by obino** [comment]
ocd run: https://ocd.appscale.com:8080/job/Daily%20Build/539/

File: AppController/djinn.rb [hunk 2]
```diff
@@ -4722,6 +4727,12 @@ def setup_appengine_application(app, is_new_app)
       # deploys?
       if is_new_app
         @num_appengines.times { |index|
+          appengine_port = find_lowest_free_port(STARTING_APPENGINE_PORT)
+          if appengine_port < 0
+            Djinn.log_warn("Failed to get port application #{app} on " +
```
Comment (menivaitsi) [inline AppController/djinn.rb]: "port application #{app}" -> "port used by #{app}"

File: AppController/lib/helperfunctions.rb [hunk 3]
```diff
@@ -1368,6 +1369,10 @@ def self.log_obscured_env()
     Djinn.log_debug(env)
   end
 
+  def self.get_avg_load()
+    return IO.read('/proc/loadavg').split[0].to_i
```
Comment (menivaitsi) [inline AppController/lib/helperfunctions.rb]: Can you make this a constant instead of hardcoding?

**Comment by obino** [comment]
OCD run: https://ocd.appscale.com:8080/job/Daily%20Build/540/

**Pull request merged**